src/lua/src/main.c

```diff
@@ -0,0 +1,186 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "lua.h"
+#include "lauxlib.h"
+#include "lualib.h"
+
+static lua_State *globalL = NULL;
+static const char *progname = NULL;
+
+static void lstop (lua_State *L, lua_Debug *ar)
+{
+	(void)ar;  /* unused arg. */
+	lua_sethook(L, NULL, 0, 0);
+	luaL_error(L, "interrupted!");
+}
+
+static void laction (int i)
+{
+	signal(i, SIG_DFL); 
+	lua_sethook(globalL, lstop, LUA_MASKCALL | LUA_MASKRET | LUA_MASKCOUNT, 1);
+}
+
+static void l_message (const char *pname, const char *msg)
+{
+	if (pname) fprintf(stderr, "%s: ", pname);
+	fprintf(stderr, "%s\n", msg);
+	fflush(stderr);
+}
+
+static int report(lua_State *L, int status)
+{
+	if (status && !lua_isnil(L, -1)) {
+		const char *msg = lua_tostring(L, -1);
+		if (msg == NULL) msg = "(error object is not a string)";
+		l_message(progname, msg);
+		lua_pop(L, 1);
+	}
+	return status;
+}
+
+static int traceback (lua_State *L)
+{
+	if (!lua_isstring(L, 1))  /* 'message' not a string? */
+		return 1;  /* keep it intact */
+	lua_getfield(L, LUA_GLOBALSINDEX, "debug");
+	if (!lua_istable(L, -1)) {
+		lua_pop(L, 1);
+		return 1;
+	}
+	lua_getfield(L, -1, "traceback");
+	if (!lua_isfunction(L, -1)) {
+		lua_pop(L, 2);
+		return 1;
+	}
+	lua_pushvalue(L, 1);  /* pass error message */
+	lua_pushinteger(L, 2);  /* skip this function and traceback */
+	lua_call(L, 2, 1);  /* call debug.traceback */
+	return 1;
+}
+
+static int docall (lua_State *L, int narg, int clear)
+{
+	int status;
+	int base = lua_gettop(L) - narg;  /* function index */
+	lua_pushcfunction(L, traceback);  /* push traceback function */
+	lua_insert(L, base);  /* put it under chunk and args */
+	signal(SIGINT, laction);
+	status = lua_pcall(L, narg, (clear ? 0 : LUA_MULTRET), base);
+	signal(SIGINT, SIG_DFL);
+	lua_remove(L, base);  /* remove traceback function */
+	/* force a complete garbage collection in case of errors */
+	if (status != 0) lua_gc(L, LUA_GCCOLLECT, 0);
+	return status;
+}
+
+static int dolibrary (lua_State *L, const char *name, int clear)
+{
+	lua_getglobal(L, "require");
+	lua_pushstring(L, name);
+	return report(L, docall(L, 1, clear));
+}
+
+struct Smain {
+	int argc;
+	char **argv;
+	int status;
+};
+
+static void pushargv(lua_State *L, char **argv, int argc, int offset)
+{
+	int i, j;
+	lua_createtable(L, argc, 0);
+	for (i = offset, j = 1; i < argc; i++, j++) {
+		lua_pushstring(L, argv[i]);
+		lua_rawseti(L, -2, j);
+	}
+}
+
+static int find_libbcc(lua_State *L)
+{
+	const char *libbcc = getenv("LIBBCC_SO_PATH");
+	char buffer[4096];
+
+	if (libbcc == NULL) {
+		char *dirname;
+
+		if (readlink("/proc/self/exe", buffer, sizeof(buffer)) < 0)
+			return -1;
+
+		dirname = strrchr(buffer, '/');
+		if (dirname == NULL)
+			return -1;
+
+		strcpy(dirname + 1, "libbcc.so");
+		libbcc = buffer;
+	}
+
+	if (access(libbcc, F_OK|R_OK|X_OK) != 0)
+		return -1;
+
+	lua_pushstring(L, libbcc);
+	lua_setglobal(L, "LIBBCC_SO_PATH");
```

Comment:
yep, my merge static libs stuff is not necessary for static bcc-lua.
libbcc.a has only bcc bits and none of clang/llvm.
so for building tests/cc/test_static or static bcc-lua it is fine,
but any external repo that wants to use bcc c++/c api it needs to be a part of bcc cmake system, otherwise clang/llvm dependency is lost.